### DISCUSSION TITLE
`clear!` function is throwing an error when visible-messages is used as well

### DISCUSSION THREAD

**Discussion by alixbrunsperger** [mhuebert/inside-out#11] [Q&A] [answered]
Hello,

We are trying to build a form with inside-out.
As the form is still displayed after we submit it, we wanted to use the `clear!` function to reset the values, but this throws an error
```
Uncaught Error: No protocol method IDeref.-deref defined for type null:
```

Here is an example of code to reproduce the behaviour :
```cljs
(defn reproducer
  []
  (with-form [foo {:bar ?bar}]
    [:<>
     [:span (str @foo)]
     [:span (forms/visible-messages foo)]
     [:input {:on-change #(reset! ?bar (.. % -target -value))
              :type :text
              :value @?bar}]
     [:input {:type :button
              :value "clear"
              :on-click #(forms/clear! foo)}]]))
```

The code above throws the error when the button is clicked. However, if you comment the line ` [:span (forms/visible-messages foo)]` then it works as expected.

This looks like a bug. What do you think about this ?

Thank you very much for your help,

**Comment by mhuebert** [comment, marked as answer]
@alixbrunsperger thanks for the report! This was indeed a bug. Should be fixed now c71bc35daa7302f5e5ca4ae683046674520cbd9d.

The bug was introduced when I optimised message-computation to happen inside a reaction (so that messages are cached and recomputed when necessary).

**Reply by alixbrunsperger** [reply]
I can confirm that the issue isn't there anymore !
Thank a lot for the quick answer and fix
